sgx-types/src/secs.rs

```diff
@@ -66,8 +66,16 @@ testaso! {
 }
 
 impl Secs {
-    /// TODO: The max size of an enclave should come from CPUID, which is inaccessible from within SGX.
-    pub const SIZE_MAX: u64 = 0x1_000_000_000;
+    /// Returns the maximum enclave size for 64bit in bytes.
+    /// CPUID.(EAX=12H, ECX=0H) enumerates Intel SGX capability;
+    /// For more on CPUID enumeration leaves, see 37.7.2 and Table 37-4.
+    pub fn max_size_64bit() -> Option<u64> {
+        let leaf = 0x00000012;
+        let subleaf = 0x00000000;
+        let res = unsafe { core::arch::x86_64::__cpuid_count(leaf, subleaf) };
+        let max_size_64bit: u64 = 1 << (res.edx >> 8 as u8) as u64;
```

Comment:
You can't just call CPUID on this leaf/subleaf. This will crash on CPUs that are not Intel or do not support sgx. So you need to test those leafs first.

Reply:
That's a good point.